File: src/main/java/nuitinfo/whiletrue/bot/Bot.java
package nuitinfo.whiletrue.bot;

import org.telegram.telegrambots.ApiContextInitializer;
import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.TelegramBotsApi;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiRequestException;

public class Bot extends TelegramLongPollingBot {

    private final String token;

    private Bot(String token) {
        this.token = token;
        System.out.println("Lancement du bot, token: '" + token + "'.");
    }

    public static void main(String[] args) {
        System.out.println("Configuration....");
        ApiContextInitializer.init();

        if(args.length == 1) {
            try {
                new TelegramBotsApi().registerBot(new Bot(args[0].replace("\"", "")));
            } catch (TelegramApiRequestException e) {
                throw new RuntimeException("Couldn't launch the API...", e);
            }
        }
        else
            System.out.println("Un seul paramètre était attendu, " + args.length + " ont été reçus.");
    }

    @Override
    public void onUpdateReceived(Update update) {
        System.out.println(update.toString());
    }

    @Override
    public String getBotUsername() {
        return "whiletrueinfobot";
    }

    @Override
    public String getBotToken() {
        return token;
    }
}

File: src/main/kotlin/nuitinfo/whiletrue/bot/Bot.kt
package nuitinfo.whiletrue.bot

import org.telegram.telegrambots.ApiContextInitializer
import org.telegram.telegrambots.bots.TelegramLongPollingBot
import org.telegram.telegrambots.meta.TelegramBotsApi
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.exceptions.TelegramApiRequestException
import java.io.File

class Bot private constructor(private val token: String) : TelegramLongPollingBot() {

    val database: Database
    val mamie: Database

    val ressources = File("src/main/resources/")
    operator fun File.get(name: String) = File(this, name)

    init {
        println("Analyse des bases de données....")
        database = Database(ressources["db.json"])
        mamie = Database(ressources["mamie.json"])

        println("Lancement du bot, token: '$token'.")
    }

    override fun onUpdateReceived(update: Update) {
        println(update.message?.date.toString() + "> " + update.message?.contact?.lastName + ": " + update.message?.text)

        update.message?.text?.let {
            database[it](this, update, database)
        }
    }

    override fun getBotUsername(): String {
        return "whiletrueinfobot"
    }

    override fun getBotToken(): String {
        return token
    }

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            println("Configuration....")
            ApiContextInitializer.init()

            if (args.size == 1) {
                try {
                    TelegramBotsApi().registerBot(Bot(args[0].replace("\"", "")))
                } catch (e: TelegramApiRequestException) {
                    throw RuntimeException("Couldn't launch the API...", e)
                }
            } else
                println("Un seul paramètre était attendu, " + args.size + " ont été reçus.")
        }
    }
}
